pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -0,0 +1,341 @@
+package usersignup
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/config"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"testing"
+)
+
+const (
+	nameMember = "east"
+)
+
+func TestReadUserApprovalPolicy(t *testing.T) {
```

Comment:
could you please use `// given` `// when` `// then` comments? with the comments it's much easier to read and understand the tests

Reply:
@MatousJobanek what do you think we should set the status to if the MUR already exists?

Reply:
@MatousJobanek I've prepared a flowchart documenting the user provisioning process, you should receive an e-mail link for this.  It would be great to get your input on the provisioning workflow, as there are still some gray areas.

Reply:
> what do you think we should set the status to if the MUR already exists?

good question :-). The goal is to create MUR, but it is already created, so from the business point of view we are done - the status can be `completed = true` with the `reason = AlreadyExists`
From the technical point of view there is a conflict (that shouldn't really happen in the ideal world), which means that we should fail and set the status to `completed = false` with the `reason = AlreadyExists/Conflict`

Unfortunatelly, we don't live in an ideal world, so I would go with the second option and let the administrator deal with the errors. In this case, you can also reque the request, but with a longer timeout.

Reply:
The second option seems reasonable to me, @alexeykazakov would you concur with this?

Reply:
Going with the `AlreadyExists` option, changes merged.

Reply:
It depends on what can trigger a reconcile. It's not safe to assume that a reconcile can happen only once when the corresponding MUR does not exist yet. Remember that the controller is supposed to match the desired state with the actual one. So, if the corresponding MUR already exists then it's not an error. Just do nothing and update the status if needed (to complete).

@sbryzak @MatousJobanek take a look at this blog - https://hackernoon.com/level-triggering-and-reconciliation-in-kubernetes-1f17fe30333d
It's a good read Xavier shared earlier.

Reply:
@alexeykazakov I'm still mulling over this one, I'll reply when I have done some more contemplating. :)